Subject: [PATCH] Fix keyguard is black background while screen turning off/on quickly

 1.the screen will be turning off an on quickly while we multi-press power
key fastly.
 2.the screen is to turned on ,but the mColorFadeOffAnimator is still animating
 3.that will not setScreenState to Display.STATE_OFF,and the next step that
screen turning on will not block screen to drawn window
 4.when the screen is on ,the keyguard interface is black background

diff --git a/services/core/java/com/android/server/display/DisplayPowerController.java b/services/core/java/com/android/server/display/DisplayPowerController.java
index e9fdcf3..b52ad5e 100644
--- a/services/core/java/com/android/server/display/DisplayPowerController.java
+++ b/services/core/java/com/android/server/display/DisplayPowerController.java
@@ -906,13 +906,16 @@
 
     private void animateScreenStateChange(int target, boolean performScreenOffTransition) {
         // If there is already an animation in progress, don't interfere with it.
-        if (mColorFadeOnAnimator.isStarted()
-                || mColorFadeOffAnimator.isStarted()) {
+        if (mColorFadeOnAnimator.isStarted()) {
+            if (target == Display.STATE_ON) {
+                // If display state changed to on, proceed and stop the color fade and turn screen on.
+                mPendingScreenOff = false;
+                return;
+            }
+        }else if (mColorFadeOffAnimator.isStarted()){
             if (target != Display.STATE_ON) {
                 return;
             }
-            // If display state changed to on, proceed and stop the color fade and turn screen on.
-            mPendingScreenOff = false;
         }
 
         // If we were in the process of turning off the screen but didn't quite
